test/functional/eval/input_spec.lua

```diff
@@ -1,38 +1,334 @@
 local helpers = require('test.functional.helpers')(after_each)
 local Screen = require('test.functional.ui.screen')
 
+local eq = helpers.eq
+local wait = helpers.wait
 local feed = helpers.feed
+local meths = helpers.meths
 local clear = helpers.clear
+local source = helpers.source
 local command = helpers.command
+local exc_exec = helpers.exc_exec
 
 local screen
 
 before_each(function()
   clear()
   screen = Screen.new(25, 5)
   screen:attach()
+  source([[
+    hi Test ctermfg=Red guifg=Red term=bold
+    function CustomCompl(...)
+      return 'TEST'
+    endfunction
+    function CustomListCompl(...)
+      return ['FOO']
+    endfunction
+  ]])
+  screen:set_default_attr_ids({
+    EOB={bold = true, foreground = Screen.colors.Blue1},
+    T={foreground=Screen.colors.Red},
+  })
 end)
 
 describe('input()', function()
   it('works correctly with multiline prompts', function()
     feed([[:call input("Test\nFoo")<CR>]])
     screen:expect([[
-      {1:~                        }|
-      {1:~                        }|
-      {1:~                        }|
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {EOB:~                        }|
       Test                     |
       Foo^                      |
-    ]], {{bold=true, foreground=Screen.colors.Blue}})
+    ]])
   end)
   it('works correctly with multiline prompts and :echohl', function()
-    command('hi Test ctermfg=Red guifg=Red term=bold')
     feed([[:echohl Test | call input("Test\nFoo")<CR>]])
     screen:expect([[
-      {1:~                        }|
-      {1:~                        }|
-      {1:~                        }|
-      {2:Test}                     |
-      {2:Foo}^                      |
-    ]], {{bold=true, foreground=Screen.colors.Blue}, {foreground=Screen.colors.Red}})
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {T:Test}                     |
+      {T:Foo}^                      |
+    ]])
+    wait()
+    command('redraw!')
+    wait()
+    screen:expect([[
+                               |
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {T:Foo}^                      |
+    ]])
+  end)
+  it('works correctly with multiple numeric arguments (many args)', function()
+    command('hi Test ctermfg=Red guifg=Red term=bold')
+    feed([[:echohl Test | call input(1, 2)<CR>]])
+    wait()  -- Without wait() it first shows `12` line and then empty line.
+    command('redraw!')  -- Without this it shows two `12` lines.
+    wait()
+    -- None of the above problems happen when testing manually.
+    screen:expect([[
+                               |
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {EOB:~                        }|
+      {T:1}2^                       |
+    ]])
+    feed('<BS>')
+    wait()
```

Comment:
in fact I can delete every single `wait()` related to screen testing and it still passes.